src/QueryBuilder/QueryBuilder/QueryBuilderModal/SelectionContainer/SelectionContainer.js

```diff
@@ -18,11 +18,22 @@ export const SelectionContainer = ({
   source,
   testId,
   emptyMessage,
+  value,
   ...rest
 }) => {
   const intl = useIntl();
   const { setDataOptions } = useContext(RootContext);
   const [searchValue, setSearchValue] = useState('');
+  const isBooleanField = availableValues?.every(opt => typeof opt.value === 'boolean');
+  let normalizedValue = value;
+  if (isBooleanField && typeof value === 'string') {
+    if (value.toLowerCase() === 'true') {
+      normalizedValue = true;
+    } else if (value.toLowerCase() === 'false') {
+      normalizedValue = false;
+    }
```

Comment:
Could we get away with just
```suggestion
    normalizedValue = value.toLowerCase() === 'true';
```
?

Reply:
Yep, good point. I originally was doing the full if-else block in case there was a value that wasn't 'true' or 'false', but if that happens, we probably have bigger problems happening elsewhere.